scripts-dev/release.py

```diff
@@ -0,0 +1,238 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright 2020 The Matrix.org Foundation C.I.C.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""An interactive script to walk through the initial stages of creating a
+release, including creating release branch, updating changelog and pushing to
+GitHub.
+"""
+
+import subprocess
+import sys
+from typing import Optional
+
+import click
+import git
+from packaging import version
+from redbaron import RedBaron
```

Comment:
the problem with this stuff is that it has to be installed in the virtualenv. We can probably add them to the dev dependencies if they are useful.

Reply:
Hmm, you're right. I'm not a huge fan of adding them as a dev dependency, but I don't have a better idea and they all make the script easier and remove boiler plate.

Reply:
I've added it under a new `"dev"` section, not sure if that's the right way to do it though.